### PR TITLE
plat-poplar: include internal test pTA

### PR DESCRIPTION
**Pull request by igoropaniuk**
```
Set CFG_TEE_CORE_EMBED_INTERNAL_TESTS, which is needed for
SDP TA to pTA tests.

Signed-off-by: Igor Opaniuk <igor.opaniuk@gmail.com>
```

### CODE REVIEW

**Review by etienne-lms — COMMENTED** [review]
s/Set/Default enable/ in commit log.
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by igoropaniuk** [comment]
@etienne-lms comment address, A-b tag applied

File: core/arch/arm/plat-poplar/conf.mk [hunk 1]
```diff
@@ -29,4 +29,6 @@ CFG_SECURE_DATA_PATH ?= y
 CFG_TEE_SDP_MEM_BASE ?= 0x02800000
 CFG_TEE_SDP_MEM_SIZE ?= 0x00400000
 
+CFG_TEE_CORE_EMBED_INTERNAL_TESTS ?= y
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-poplar/conf.mk:32]: Any reason why you're not assigning `CFG_ENABLE_EMBEDDED_TESTS` instead?
With that you'd catch other future embedded tests too.

Comment (igoropaniuk) [reply core/arch/arm/plat-poplar/conf.mk:32]: Not really, I followed the same way as it's done for other boards:

```
$ grep -e CFG_TEE_CORE_EMBED_INTERNAL_TESTS -r ./core/arch/arm/plat-*
./core/arch/arm/plat-rpi3/conf.mk:CFG_TEE_CORE_EMBED_INTERNAL_TESTS ?= y
./core/arch/arm/plat-stm/conf.mk:CFG_TEE_CORE_EMBED_INTERNAL_TESTS ?= y
./core/arch/arm/plat-stm32mp1/conf.mk:CFG_TEE_CORE_EMBED_INTERNAL_TESTS ?= y


```

Comment (igoropaniuk) [reply core/arch/arm/plat-poplar/conf.mk:32]: So probably it makes sense to `s/CFG_TEE_CORE_EMBED_INTERNAL_TESTS/CFG_ENABLE_EMBEDDED_TESTS/g` for other boards

Comment (jenswi-linaro) [reply core/arch/arm/plat-poplar/conf.mk:32]: I should have mentioned this earlier.
The reason we have this default off for everything else is to minimize the risk that debug or test code is unintentionally included in a product. So you may want to set `CFG_ENABLE_EMBEDDED_TESTS=y` in the build script you use or so.
With that in mind, if you really want it here I suppose that's OK too.

Comment (igoropaniuk) [reply core/arch/arm/plat-poplar/conf.mk:32]: Got it. Let's do that differently then.
Please ignore this PR, and I'll add `CFG_ENABLE_EMBEDDED_TESTS=y` to https://github.com/OP-TEE/build/blob/master/poplar.mk. We will kill two birds with one stone: make xtest running without errors (in our default buildroot setup) and avoid default enabling these tests in OP-TEE core

Comment (jenswi-linaro) [reply core/arch/arm/plat-poplar/conf.mk:32]: Sound good.

**Pull request closed without merging**